src/Tests/Essential.Diagnostics.RollingFileTraceListener.Tests/Utility/MockFileSystem.cs

```diff
@@ -7,6 +7,8 @@ namespace Essential.Diagnostics.Tests.Utility
 {
     public class MockFileSystem : IFileSystem
     {
+        public bool CreateSubdirectories { get; set; }
```

Comment:
As mentioned above, mock out the GetDirectoryName (can just forward to the real one) and CreateDirectory (keep a list of created) methods, and use them to test the logic.